plugins/optimization-detective/detect.js

```diff
@@ -507,6 +510,20 @@ export default async function detect( {
 		);
 	} );
 
+	// Only proceed with submitting the URL Metric if viewport stayed the same size. Changing the viewport size (e.g. due
+	// to resizing a window or changing the orientation of a device) will result in unexpected metrics being collected.
+	if (
+		window.innerWidth !== urlMetric.viewport.width ||
+		window.innerHeight !== urlMetric.viewport.height
+	) {
+		if ( isDebug ) {
+			log(
+				'Aborting URL Metric collection due to viewport size change.'
+			);
+		}
+		return;
+	}
```

Comment:
From @felixarntz in https://github.com/WordPress/performance/pull/1697#discussion_r1866399258:

> Couldn't this be moved much further up in the logic, given we'll know whether the viewport has changed far earlier in this function? This could avoid doing quite a bit of "useless" processing as far as I can tell.

Reply:
Awaiting an answer to this question, otherwise LGTM :)

Reply:
@felixarntz Does this make sense?

Reply:
@westonruter Makes sense. I wonder though, would there be other tasks here that could be deferred to when leaving the page so that they don't run unnecessarily? For example some of the LCP element logic that runs before then as far as I can see?

Reply:
Yes, although I think if we wait until this point to call `onLCP()` I'd worry about increasing the amount of time needing to be spent at this `pagehide` phase and risk the browser killing the tasks before it gets a chance to send the beacon. But I think that is something that we can investigate further in a follow-up.